### PR TITLE
refactor(jest): convert jest mocks to msw

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

We have at least four different ways of mocking the API:
- Jest tests with custom mocking code
- Storybook scenes with `mockserviceworker` aka `msw`
- Storybook scenes with captured redux state
- Cypress tests with custom fixtures
- ...

Some more context here: https://github.com/PostHog/posthog/issues/6589

## Changes

- This PR is the first step in unifying the different mocks throughout the app, with the goal of eventually having a great developer experience when working with storybook.

- This PR migrates jest tests to use `msw` with a few wrappers. This is instead of the old `api.get` mocking we had before. 

- This PR also gets rid of the legacy `initKeaTestLogic` code and replaces it with `initKeaTests` everywhere.

- The new mocks can be used anywhere in jest tests: in all `test`, `it` or `describe` blocks. If you override an API call lower down the hierarchy, it'll be properly overridden. 

- There's a file, `handlers.ts` that contains the default mocks, which will be shared between jest, cypress and storybook.

- I also created a function `useFeatures()` that lets you enable/disable features returned in the default mocks.

The code now looks like this:

```ts
beforeEach(() => {
    useFeatures([AvailableFeature.CORRELATION_ANALYSIS, AvailableFeature.GROUP_ANALYTICS])
    useMocks({
        get: {
            '/api/projects/:team/insights': { results: ['result from api'] },
            '/api/projects/:team/insights/123': { result: ['result from api'] },
            '/api/projects/:team/insights/trend/': (req) => ({ result: [req.url.searchParams.get('insight_id')] }),
        },
    })
    initKeaTests()
})
```

## Next steps

- Migrate redux state in storybook and everything else to `msw` as well
- I had to `yeet` one test in `eventDefinitionsTableLogic` that was awkwardly built with too deep mocks. This should be fixed.

## How did you test this code?

- These are literally tests. They all ran.

### CODE REVIEW

**Comment by mariusandra** [comment]
While all tests run perfectly locally, something is weird in CI. Still since there are so many changed files, I'd appreciate if someone could take a look while I fix the last things.

**Review by pauldambra — CHANGES_REQUESTED** [review]
I like the change to using beforeEach for everything 🙌 

running `yarn test` three times on master takes me about 17 seconds each, on this branch 20 seconds each. May be coincidence, certainly not much worse

I get two failures in Cypress but would take 6 minutes to see if that's related to this branch by running in master too 🤣

File: frontend/src/mocks/handlers.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        ],
+        '/api/users/@me/': () => [
+            200,
+            {
+                organization: { ...MOCK_DEFAULT_ORGANIZATION, available_features: getFeatures() },
+                team: MOCK_DEFAULT_TEAM,
+            },
+        ],
+        '/api/projects/@current/': MOCK_DEFAULT_TEAM,
+        '/_preflight': require('../../../cypress/fixtures/_preflight.json'),
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/mocks/handlers.ts]: Ideally the files should be shared between the two tests, but that's a TODO. Duplicated to a local folder for now.

File: frontend/src/scenes/funnels/funnelLogic.test.ts [hunk 2]
```diff
@@ -59,187 +58,207 @@ describe('funnelLogic', () => {
...
+                    200,
+                    {
+                        ...MOCK_DEFAULT_TEAM,
+                        correlation_config: correlationConfig,
+                    },
+                ],
+                '/api/projects/:team/insights/': { results: funnelResults },
+                '/api/projects/:team/insights/trend/': { results: ['trends result from api'] },
+                '/api/projects/:team/groups_types/': [
+                    // { group_type: 'project', group_type_index: 0, name_singular: null, name_plural: null },
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/scenes/funnels/funnelLogic.test.ts]: Ragecomment. Removed.

**Review by Twixes — APPROVED** [review]
Overall looks good (just +1 to Paul's comments). Also, `useFeatures`/`useMocks`/etc. seem very React hook-like to me, which is kind of different from the semantics here, but that doesn't really matter.

**Pull request merged**